cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/LedgerEvents/Gov/ProposeNewConstitutionSPO.hs

```diff
@@ -0,0 +1,248 @@
+  gov <- H.createDirectoryIfMissing $ work </> "governance"
+  proposalAnchorFile <- H.note $ work </> gov </> "sample-proposal-anchor"
+  constitutionFile <- H.note $ work </> gov </> "sample-constitution"
+  constitutionActionFp <- H.note $ work </> gov </> "constitution.action"
+
+  H.writeFile proposalAnchorFile "dummy anchor data"
+  H.writeFile constitutionFile "dummy constitution data"
+  constitutionHash <- H.execCli' execConfig
+    [ "conway", "governance"
+    , "hash", "anchor-data", "--file-text", constitutionFile
+    ]
+
+  proposalAnchorDataHash <- H.execCli' execConfig
+    [ "conway", "governance"
+    , "hash", "anchor-data", "--file-text", proposalAnchorFile
+    ]
+
+  let stakeVkeyFp = gov </> "stake.vkey"
+      stakeSKeyFp = gov </> "stake.skey"
+
+  _ <- P.cliStakeAddressKeyGen tempAbsPath'
+         $ P.KeyNames { P.verificationKeyFile = stakeVkeyFp
+                      , P.signingKeyFile = stakeSKeyFp
+                      }
+
+  let spoColdVkeyFp :: Int -> FilePath
+      spoColdVkeyFp n = tempAbsPath' </> "pools-keys" </> "pool" <> show n </> "cold.vkey"
+
+      spoColdSkeyFp :: Int -> FilePath
+      spoColdSkeyFp n = tempAbsPath' </> "pools-keys" </> "pool" <> show n </> "cold.skey"
+
+  -- Create constitution proposal
+  void $ H.execCli' execConfig
+    [ "conway", "governance", "action", "create-constitution"
+    , "--testnet"
+    , "--governance-action-deposit", show @Int 0 -- TODO: Get this from the node
+    , "--deposit-return-stake-verification-key-file", stakeVkeyFp
+    , "--anchor-url", "https://tinyurl.com/3wrwb2as"
+    , "--anchor-data-hash", proposalAnchorDataHash
+    , "--constitution-url", "https://tinyurl.com/2pahcy6z"
+    , "--constitution-hash", constitutionHash
+    , "--out-file", constitutionActionFp
+    ]
+
+  txbodyFp <- H.note $ work </> "tx.body"
+  txbodySignedFp <- H.note $ work </> "tx.body.signed"
+
+  txin1 <- queryAnyUtxo . paymentKeyInfoAddr $ head wallets
+
+  void $ H.execCli' execConfig
+    [ "conway", "transaction", "build"
+    , "--testnet-magic", show @Int testnetMagic
+    , "--tx-in", Text.unpack $ renderTxIn txin1
+    , "--change-address", Text.unpack $ paymentKeyInfoAddr $ head wallets
+    , "--proposal-file", constitutionActionFp
+    , "--out-file", txbodyFp
+    ]
+
+  void $ H.execCli' execConfig
+    [ "conway", "transaction", "sign"
+    , "--testnet-magic", show @Int testnetMagic
+    , "--tx-body-file", txbodyFp
+    , "--signing-key-file", paymentSKey $ paymentKeyInfoPair $ head wallets
+    , "--out-file", txbodySignedFp
+    ]
+
+  void $ H.execCli' execConfig
+    [ "conway", "transaction", "submit"
+    , "--testnet-magic", show @Int testnetMagic
+    , "--tx-file", txbodySignedFp
+    ]
+
+  txidString <- mconcat . lines <$> H.execCli' execConfig
+    [ "transaction", "txid"
+    , "--tx-file", txbodySignedFp
+    ]
+  !propSubmittedResult
+    <- runExceptT $ handleIOExceptT IOE
+                  $ runExceptT $ foldBlocks
```

Comment:
`checkLedgerState` is preferred way of condition checking now whenever it's possible.